### PR TITLE
feat: Added support for the unit type ('()').

### PR DESCRIPTION
**Pull request by ckaran**
Added support for the unit (`()`) type, which is a zero-sized type.  As you'd expect, this neither reads nor writes to the underlying storage.  The main purpose for this is for derive macros which may require that all elements of a struct also implement the various traits.  Basically, this is a partial answer to #9.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,6 +92,13 @@ Layouts are defined using the [define_layout!](https://docs.rs/binary-layout/lat
 2. The [FieldView](https://docs.rs/binary-layout/latest/binary_layout/struct.FieldView.html) API that wraps a slice of storage data and remembers it in a `View` object, allowing access to the fields without having to pass in the packed data slice each time. This is the API used in the example above. See [FieldView](https://docs.rs/binary-layout/latest/binary_layout/struct.FieldView.html) for another example.
 
 ### Supported field types
+
```
Comment (smessmer) [inline README.md:95]: let's move this to the end of the list. Since people likely are more interested in the primitive integer types than in the ZSTs, they should see those first in the documentation.

**Review by smessmer — CHANGES_REQUESTED** [review]
Can I ask you to
- add a changelog entry
- add a unit test somewhere that includes a zero sized type in an actual `define_layout!` invocation. I could imagine parentheses screwing up the macro somehow.
